src/overlays/actors/ovl_En_Floormas/z_en_floormas.c

```diff
@@ -9,123 +15,1120 @@ void EnFloormas_Destroy(Actor* thisx, GlobalContext* globalCtx);
+void EnFloormas_Run(EnFloormas* this, GlobalContext* globalCtx);
+void EnFloormas_BigStopWalk(EnFloormas* this, GlobalContext* globalCtx);
+void EnFloormas_BigWalk(EnFloormas* this, GlobalContext* globalCtx);
+void EnFloormas_Stand(EnFloormas* this, GlobalContext* globalCtx);
+void EnFloormas_BigDecideAction(EnFloormas* this, GlobalContext* globalCtx);
+void EnFloormas_Charge(EnFloormas* this, GlobalContext* globalCtx);
+
 const ActorInit En_Floormas_InitVars = {
     ACTOR_EN_FLOORMAS,
     ACTORTYPE_ENEMY,
     FLAGS,
     OBJECT_WALLMASTER,
     sizeof(EnFloormas),
-    (ActorFunc)EnFloormas_Init,
-    (ActorFunc)EnFloormas_Destroy,
-    (ActorFunc)EnFloormas_Update,
-    (ActorFunc)EnFloormas_Draw,
+    EnFloormas_Init,
+    EnFloormas_Destroy,
+    EnFloormas_Update,
+    EnFloormas_Draw,
+};
+
+static ColliderCylinderInit cylinderInit = {
+    { COLTYPE_UNK0, 0x11, 0x09, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0xFFCFFFFF, 0x04, 0x10 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x09, 0x05, 0x01 },
+    { 25, 40, 0, { 0, 0, 0 } },
+};
+
+static CollisionCheckInfoInit colCheckInfoInit = { 0x04, 0x001E, 0x0028, 0x96 };
+
+static DamageTable damageTable = { {
+    0x10, 0x02, 0x01, 0x02, 0x10, 0x02, 0x02, 0x10, 0x01, 0x02, 0x04, 0x24, 0x02, 0x44, 0x04, 0x02,
+    0x02, 0x24, 0x00, 0x44, 0x00, 0x00, 0x01, 0x04, 0x02, 0x02, 0x08, 0x04, 0x00, 0x00, 0x04, 0x00,
+} };
+
+static InitChainEntry initChain[] = {
+    ICHAIN_S8(naviEnemyId, 0x31, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 0x157C, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, 0xFC18, ICHAIN_STOP),
+};
+
+static Vec3f D_80A1A4D0 = {
+    0.0f,
+    0.0f,
+    0.0f,
+};
+
+static Vec3f D_80A1A4DC = {
+    0.0f,
+    0.0f,
+    0.0f,
+};
+
+static Vec3f dustPos = {
+    0.0f,
+    0.0f,
+    0.0f
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/EnFloormas_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/EnFloormas_Destroy.s")
+static Color_RGBA8 mergeColor = {
+    0x00,
+    0xFF,
+    0x00,
+    0x00,
+};
+
+// display lists
+extern Gfx D_06008688[];
+
+// skeleton
+extern SkeletonHeader D_06008FB0;
+
+// animations
+extern AnimationHeader D_06009DB0;
+extern AnimationHeader D_060039B0;
+extern AnimationHeader D_06000EA4;
+extern AnimationHeader D_06000590;
+extern AnimationHeader D_060019CC;
+extern AnimationHeader D_06009520;
+extern AnimationHeader D_06002158;
+extern AnimationHeader D_0600A054;
+extern AnimationHeader D_060041F4;
+extern AnimationHeader D_06009244;
+
+void EnFloormas_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFloormas* this = THIS;
+    GlobalContext* gctx = globalCtx;
+    s32 invisble;
+    char pad[4];
+
+    Actor_ProcessInitChain(&this->actor, initChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 50.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06008FB0, &D_06009DB0, &this->limbDrawTable, &this->transDrawTable,
```

Comment:
weve been going with transitionDrawTable everywhere in the codebase. though i wouldnt be against this shorter name, other than it not being very clear what it means at first glance. down to talk about it in discord or something